src/measures/MultiS3.cpp

```diff
@@ -0,0 +1,77 @@
+#include "MultiS3.hpp"
+#include <cmath>
+
+unsigned MultiS3::denom = 0;
+vector<uint> MultiS3::shadowDegrees;
+
+MultiS3::MultiS3(Graph* G1, Graph* G2) : Measure(G1, G2, "ms3")
+{
+    G1->printStats(0, cout);
+    G2->printStats(0, cout);
+    initDegrees(*G2);
+}
+
+MultiS3::~MultiS3() {
+}
+
+void MultiS3::initDegrees(const Graph& G2)
+{
+    shadowDegrees = vector<uint>(G2.getNumNodes(), 0);
+    
+    vector<vector<uint>> G2AdjLists;
+    G2.getAdjLists(G2AdjLists);
+    
+    Matrix<MATRIX_UNIT> G2Matrix;
+    G2.getMatrix(G2Matrix);
+    
+    const uint n = G2AdjLists.size();
+    uint neighbor;
+    
+    for (uint i = 0; i < n; ++i)
+    {
+        for (uint j = 0; j < G2AdjLists[i].size(); ++j)
+        {
+            neighbor = G2AdjLists[i][j];
+            shadowDegrees[i] += G2Matrix[i][neighbor];
+        }
+    }
+}
+
+void MultiS3::getDenom(const Alignment& A, const Graph& G1, const Graph& G2)

```

Comment:
Since it doesn't return anything, I'd call it computeDenom, or better, computeNumLadders... and why not just return the number in addition to compute the data member?